src/main/java/edu/tntech/graph/sampler/Sampler.java

```diff
@@ -0,0 +1,381 @@
+package edu.tntech.graph.sampler;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.tntech.graph.exception.SampleNotStoredException;
+import edu.tntech.graph.helper.ConfigReader;
+import edu.tntech.graph.helper.GraphHelper;
+import edu.tntech.graph.helper.Helper;
+import edu.tntech.graph.pojo.Edge;
+import edu.tntech.graph.pojo.Node;
+import edu.tntech.graph.pojo.Sample;
+import edu.tntech.graph.stream.StreamProcessor;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+public class Sampler {
+
+    private static ConfigReader config;
+
+    private Sample sample;
+
+    private Integer sampleSize;
+
+    private static Sampler instance;
+
+    private Helper helper;
+
+    private Sampler(Integer sampleSize) {
+        this.sampleSize = sampleSize;
+        this.helper = Helper.getInstance();
+
+        setSample();
+    }
+
+    public static Sampler getInstance() throws FileNotFoundException {
+        if (instance == null) {
+            config = ConfigReader.getInstance();
+            int sampleSize = Integer.parseInt(config.getProperty("sample-size"));
+            instance = new Sampler(sampleSize);
+        }
+        return instance;
+    }
+
+    public Integer getSampleSize() {
+        return sampleSize;
+    }
+
+    public Sample getSample() {
+        return sample;
+    }
+
+    public void setSample() {
+        boolean sampleStored = Boolean.parseBoolean(config.getProperty(Sample.STORE_SAMPLE_INDEX));
+        if (sampleStored) {
+            try {
+
+                Path path = Paths.get(helper.getAbsolutePath(Sample.SAMPLE_FILE));
+                if (Files.exists(path)) {
+                    sample = GraphHelper.getStoredSample();
+                }
+
+            } catch (IOException e) {
+                throw new SampleNotStoredException("Sample is not stored");
+            } catch (SampleNotStoredException e) {
```

Comment:
try block le IOException chai fyalcha .. if it does not find the file.I guess IOException catches PropertyNotFoundException pani because it compiles :D

Reply:
hoina hola...PropertyNotFoundException is runtime exception while IOException is not.

Reply:
maile chai SampleNotStoredException chai throw nai hudaina yaha bhaneko khas